sycl/unittests/scheduler/AllocaLinking.cpp

```diff
@@ -48,17 +48,17 @@ TEST_F(SchedulerTest, AllocaLinking) {
     return;
   }
 
+  device HostDevice{host_selector()};
```

Comment:
Could you please clarify why changes in this file are needed?

Reply:
I had to move that line before Mock.redefines().
Otherwise, MOck.redefines() disallowed creating host device properly.

Reply:
Could you please provide more details? Shouldn't we document this limitation?
Tagging @AGindinson @sergey-semenov

Reply:
I sent an email about why I had to move the host_device constructor.
In summary, Mock.redefine(redefinedDeviceGetInfo) overrides the real implementation of get_device_info which is needed when creating a host_device.